### PR TITLE
Add Naming & Data Models for Custom Deployment SDK

### PR DESCRIPTION
**Pull request by desertaxle**
## Summary

- Adds naming utilities for converting arbitrary names to valid Python identifiers and class names
- Adds data models for SDK generation (WorkPoolInfo, DeploymentInfo, FlowInfo, SDKData)
- Includes 227 comprehensive tests covering edge cases

### Naming Utilities (`src/prefect/_sdk/naming.py`)
- `to_identifier()`: Convert names to valid Python identifiers
- `to_class_name()`: Convert names to PascalCase class names
- Unicode separator handling (em-dash, non-breaking space become word boundaries)
- NFKD normalization for accented characters (é → e)
- Python keyword handling
- Reserved name detection for SDK surface
- Collision resolution with numeric suffixes

### Data Models (`src/prefect/_sdk/models.py`)
- `WorkPoolInfo`: Work pool name, type, job variables schema
- `DeploymentInfo`: Deployment name, flow name, parameter schema, work pool ref
- `FlowInfo`: Flow name with list of deployments
- `SDKGenerationMetadata`: Generation time, Prefect version, workspace, API URL
- `SDKData`: Complete container for SDK generation with convenience methods

See implementation notes in `plans/2026-01-06-custom-deployment-sdks.md` for deviations from the original plan.

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: plans/2026-01-06-custom-deployment-sdks.md [hunk 1]
```diff
@@ -242,14 +242,67 @@ This matches how Prefect's server validates parameters (`actions.py:287-304`).
...
+   - Rationale: Simplifies template rendering and statistics reporting
+
+9. **SDKGenerationMetadata.api_url added**
+   - Plan doesn't include this field
+   - Implementation: Added `api_url` field
+   - Rationale: Better traceability of SDK generation source
+
+10. **German ß limitation**
+    - Plan doesn't address
+    - Implementation: ß is dropped (NFKD doesn't decompose it to "ss"), so `straße` → `strae`
```
Comment (zzstoatzz) [inline plans/2026-01-06-custom-deployment-sdks.md:304]: lol

File: src/prefect/_sdk/naming.py [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+
+Conversion rules for class names:
+1. Same Unicode handling as identifiers
+2. Split on separators/punctuation to get word parts
+3. Capitalize first letter of each part (PascalCase)
+4. Prefix with underscore if starts with digit
+5. Return "_Unnamed" if result is empty
+"""
+
+import keyword
```
Comment (zzstoatzz) [inline src/prefect/_sdk/naming.py:26]: TIL

File: src/prefect/_sdk/naming.py
Comment (zzstoatzz) [inline src/prefect/_sdk/naming.py:1]: i get a little nervous about how whack-a-mole-y this feels, but i don't have a constructive suggestion

i feel fine about it though bc the module's private and it seems like it'd be easy to add edge case handling as it comes up
